packages/api2/src/settings/settings.service.ts

```diff
@@ -0,0 +1,193 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Request } from 'express';
+import { Model, Types } from 'mongoose';
+import { Settings, SettingsDocument } from './schemas/settings.schema';
+import { AxiosResponse } from 'axios';
+import axios from 'axios';
+import { TransformerMapOperateItem } from 'ses-node-json-transform';
+import { ExportTransformer } from 'src/shared/types.shared';
+import { SetSettingDto } from './dto/set-setting.dto';
+import { ServiceException } from '../shared/service-exception';
+import { UploadedFile } from 'express-fileupload';
+import mime from 'mime-types';
+import { UtilsProvider } from '@/utils/utils.provider';
+import { unlink } from 'fs/promises';
+
+@Injectable()
+export class SettingsService {
+  constructor(
+    @InjectModel(Settings.name)
+    private settingsModel: Model<SettingsDocument>,
+    private utils: UtilsProvider,
+  ) {}
+
+  private uploadDirectory =
+    process.env.NODE_ENV === 'production' ? '/usr/src/uploads/' : 'uploads/';
+
+  async findAll(): Promise<Settings[]> {
+    const settings = await this.settingsModel.find().lean();
+    return settings.map((setting) => new Settings(setting));
+  }
+
+  async findOneById(_id: Types.ObjectId): Promise<Settings> {
```

Comment:
`UserService` returns `User | null` instead of throwing error for these simple find methods. We should pick one method and use everywhere. I have no clear opinion as to what is best. In the case of `userService.findOneByEth` returning null felt ok since we expect that method to be called with unknown addresses. Not finding a user is no exception then. But.. for the sake of clarity all over, and knowing that for most other methods not finding the object *is* an exception, perhaps we should stick to throwing exceptions?

Reply:
Yeah, i know that we talked about this. I was trying both and like you said - I think maybe this is more clear because how should we decide what is the simple method and what is not? :)
I would stick with this but we can do it like it is done in userService if you like